node/rpc/src/state_api.rs

```diff
@@ -802,16 +799,144 @@ pub(crate) async fn state_miner_sector_allocated<
     Ok(allocated_sectors)
 }
 
+pub(crate) async fn state_miner_pre_commit_deposit_for_power<
+    DB: BlockStore + Send + Sync + 'static,
+    KS: KeyStore + Send + Sync + 'static,
+    B: Beacon + Send + Sync + 'static,
+    V: ProofVerifier + Send + Sync + 'static,
+>(
+    data: Data<RpcState<DB, KS, B>>,
+    Params(params): Params<(AddressJson, SectorPreCommitInfo, TipsetKeysJson)>,
+) -> Result<String, JsonRpcError> {
+    let (AddressJson(maddr), pci, TipsetKeysJson(tsk)) = params;
+    let ts = data.chain_store.tipset_from_keys(&tsk).await?;
+    let (state, _) = data.state_manager.tipset_state::<V>(&ts).await?;
+    let state = StateTree::new_from_root(data.chain_store.db.as_ref(), &state)?;
+    let ssize = pci.seal_proof.sector_size()?;
+
+    let actor = state
+        .get_actor(market::ADDRESS)?
+        .ok_or("couldnt load market actor")?;
+    let (w, vw) = market::State::load(data.state_manager.blockstore(), &actor)?
+        .verify_deals_for_activation(
+            data.state_manager.blockstore(),
+            &pci.deal_ids,
+            &maddr,
+            pci.expiration,
+            ts.epoch(),
+        )?;
+    let duration = pci.expiration - ts.epoch();
+    let sector_weight = actor::actorv2::miner::qa_power_for_weight(ssize, duration, &w, &vw);
+
+    let actor = state
+        .get_actor(power::ADDRESS)?
+        .ok_or("couldnt load market actor")?;
+    let deposit = match power::State::load(data.state_manager.blockstore(), &actor)? {
+        power::State::V0(s) => {
+            let power_smoothed = s.this_epoch_qa_power_smoothed;
+            let reward_actor = state
+                .get_actor(reward::ADDRESS)?
+                .ok_or("couldnt load market actor")?;
+            reward::State::load(data.state_manager.blockstore(), &reward_actor)?
+                .pre_commit_deposit_for_power(power_smoothed.into(), &sector_weight)
+        }
+        power::State::V2(s) => {
+            let power_smoothed = s.this_epoch_qa_power_smoothed;
+            let reward_actor = state
+                .get_actor(reward::ADDRESS)?
+                .ok_or("couldnt load market actor")?;
+            reward::State::load(data.state_manager.blockstore(), &reward_actor)?
+                .pre_commit_deposit_for_power(power_smoothed.into(), &sector_weight)
+        }
```

Comment:
same here, would be better if this was done through interface, only this epoch qa power needs to be exposed

Reply:
Done. Testing out right now